### ISSUE TITLE
high resource use without containers upon booting

### ISSUE THREAD

**Issue by dseeker** [docker/for-linux#468]
* [x] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Expected behavior

Minimal cpu (<2%) load associated with Docker.service

### Actual behavior

In the first 10 minutes the services increasingly use CPU and RAM resources before "settling down" to normal operation

Netdata charts starting on boot
![netdata dashboard 2](https://user-images.githubusercontent.com/11655138/47601900-2b276c00-d9d7-11e8-96ae-0325ce8f7b5e.png)

After the first CPU/RAM peak
![netdata dashboard 3](https://user-images.githubusercontent.com/11655138/47601919-5e69fb00-d9d7-11e8-8307-e254868ce225.png)

Settling down
![netdata dashboard 4](https://user-images.githubusercontent.com/11655138/47601922-65910900-d9d7-11e8-8514-4a9d5a97cc9f.png)

### Steps to reproduce the behavior

Just installed Docker and run 1 container as test

**Output of `docker version`:**

```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Output of `docker info`:**

```
Containers: 5283
 Running: 0
 Paused: 0
 Stopped: 5283
Images: 2
Server Version: 18.06.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-38-generic
Operating System: Ubuntu 18.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 12
Total Memory: 62.84GiB
Name: *****
ID: JRJ7:2M7Z:4AHH:CS7M:6PYT:WFV3:S76M:RXKS:CDOL:KM43:RYQ7:TV2S
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support

```

**Additional environment details (AWS, VirtualBox, physical, etc.)**

Hetzner EX61 physical server

Service Status
```
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: active (running) since Sat 2018-10-27 10:41:49 CEST; 18min ago
     Docs: https://docs.docker.com
 Main PID: 1602 (dockerd)
    Tasks: 5287
   CGroup: /system.slice/docker.service
           ├─1602 /usr/bin/dockerd -H fd://
           └─1968 docker-containerd --config /var/run/docker/containerd/containerd.toml

Oct 27 10:32:10  dockerd[1602]: time="2018-10-27T10:32:10.476103798+02:00" level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Oct 27 10:32:10  dockerd[1602]: time="2018-10-27T10:32:10.476205098+02:00" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc42044ae90, CONNECTING" module=grpc
Oct 27 10:32:10  dockerd[1602]: time="2018-10-27T10:32:10.476499302+02:00" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc42044ae90, READY" module=grpc
Oct 27 10:32:10  dockerd[1602]: time="2018-10-27T10:32:10.476537995+02:00" level=info msg="Loading containers: start."
Oct 27 10:38:35 dockerd[1602]: time="2018-10-27T10:38:35.641924932+02:00" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address"
Oct 27 10:41:43 dockerd[1602]: time="2018-10-27T10:41:43.227249434+02:00" level=info msg="Loading containers: done."
Oct 27 10:41:49 dockerd[1602]: time="2018-10-27T10:41:49.224608316+02:00" level=info msg="Docker daemon" commit=e68fc7a graphdriver(s)=overlay2 version=18.06.1-ce
Oct 27 10:41:49 dockerd[1602]: time="2018-10-27T10:41:49.239887949+02:00" level=info msg="Daemon has completed initialization"
Oct 27 10:41:49 systemd[1]: Started Docker Application Container Engine.
Oct 27 10:41:49 dockerd[1602]: time="2018-10-27T10:41:49.378428502+02:00" level=info msg="API listen on /var/run/docker.sock"
```

**Comment by thaJeztah** [comment]
I do see you have a lot of stopped containers still on that machine (5283); does the resource usage still reproduce if you remove those containers? (running `docker container prune` should remove all stopped containers) 

There's a similar issue, but in that case with many images (https://github.com/moby/moby/issues/31462)

**Comment by dseeker** [comment]
I had disabled boot autostart for docker and after a fresh boot I executed `docker container prune`, terminal outputs nothings and it's been like this for the past 10 minutes. Memory keeps going up, CPU is seems stuck in a loop
![netdata dashboard](https://user-images.githubusercontent.com/11655138/48292521-f60d2600-e47a-11e8-9f4d-50b42f474ff6.png)

**Comment by dseeker** [comment]
![image](https://user-images.githubusercontent.com/11655138/48299284-59bf3f80-e4cb-11e8-9fc7-551e6fb087f2.png)
for the past half hour
![image](https://user-images.githubusercontent.com/11655138/48299276-3b594400-e4cb-11e8-8c00-16ab9ba56aa4.png)

**Comment by thaJeztah** [comment]
The high resource use during `docker container prune` may be something to look into. Wondering if it's related to the output not being flushed until the command completes.

Did the startup improve after those containers have been removed? (and does `docker info` show that the containers have been removed succesfully?)

**Comment by dseeker** [comment]
@thaJeztah yes it appear to have fixed the startup issue. I will keep an eye if multiple runs of the encoder will affect this and make the problem return, thanks!

**Comment by dseeker** [comment]
seems to have come back after certain use. so something in how I'm using docker is creating this problem. 
![image](https://user-images.githubusercontent.com/11655138/48548452-44805180-e8cd-11e8-8530-1c95383610d7.png)

I'm basically using node.js spawn and execSync methods to run an FFMPEG container. perhaps it has to do with executing multiple in a single process? 
"docker run cmd && docker run cmd && docker run cmd"

**Comment by thaJeztah** [comment]
Each `docker run cmd` will create a new container, execute the command, and (once the command exits), exit the container.

Those containers won't be cleaner up though (unless you're using the `--rm` option, or removing the container yourself afterwards), so all those containers will continue to exist (but in an "exited" state).

If you need to run a number of commands after eachother to process the same files, you could combine those in a single `docker run`, e.g. `docker run sh -'command 1 && command 2 ....'` (or have an image that automatically executes those commands)

You can also start a container, keep it running and `docker exec` to run new processes in that container after it's started.

**Comment by swapnilpotnis** [comment]
Hi @thaJeztah , we are facing the same issue with only "ubuntu-minimal" of 18.04 along with docker being installed over the server. 
The server has around 16GB RAM and containerd seems to be using almost 9GB of RAM at times, whilst no container is running at that point of time.
Is there any bug at the docker side already open or is it something different??
Can you please help??

**Comment by thaJeztah** [comment]
@swapnilpotnis very hard to tell just from that information. Might be better to open a new ticket with details and if possible, steps to reproduce.
